### DISCUSSION TITLE
Question re sub-domain interpolation example

### DISCUSSION THREAD

**Discussion by ethrelfall** [firedrakeproject/firedrake#3402] [Firedrake support]
Dear community,

I've got a question about interpolating onto sub-domain meshes.

In the example at https://www.firedrakeproject.org/interpolation.html (the line integral example starting with def test_line_integral():) I don't get where the line mesh mentioned in  V_line = FunctionSpace(line, "CG", 2) comes from.  In the para above it says that an immersed 1D mesh is created, but where is the mesh called "line" created?

Search revealed an answer related to MeshHierarchy but I still don't see why the example contains the "line" object ... 

Thanks!

Ed

**Comment by connorjward** [comment]
The code is taken from [here](https://github.com/firedrakeproject/firedrake/blob/master/tests/regression/test_interpolation_manual.py). We have a known issue where the sample code in the docs are generated from some of our tests, but as lines are added to the file the code moves so the wrong snippets are included in the manual. This should get fixed shortly.
